app/integration_test/src/integration_test.cc

```diff
@@ -65,4 +65,43 @@ TEST_F(FirebaseAppTest, TestDefaultAppWithDefaultOptions) {
   default_app = nullptr;
 }
 
+TEST_F(FirebaseAppTest, TestGetAll) {
+  std::vector<firebase::App*> all_apps = firebase::App::GetApps();
+  EXPECT_EQ(all_apps.size(), 0);
+
+  firebase::App* default_app;
+  default_app = firebase::App::Create(APP_CREATE_PARAMS);
+
+  all_apps = firebase::App::GetApps();
```

Comment:
You can make this test much shorter and more readable by doing something like
EXPECT_THAT(firebase::App::GetApps(), UnorderedElementsAre(default_app));
// add second_app...
EXPECT_THAT(firebase::App::GetApps(), UnorderedElementsAre(default_app, second_app));

etc.